src/lint/track_config.nim

```diff
@@ -621,6 +621,39 @@ proc satisfiesSecondPass(trackConfigContents: string; path: Path): bool =
   checkExercisesPCP(practiceExercises, result, path)
   checkExerciseSlugsAndForegone(exercises, result, path)
 
+proc getExerciseSlugs(data: JsonNode; k: string): HashSet[string] =
+  result = initHashSet[string]()
+  if data.kind == JObject:
+    if data.hasKey("exercises"):
+      let exercises = data["exercises"]
+      if exercises.kind == JObject:
+        if exercises.hasKey(k):
+          if exercises[k].kind == JArray:
+            for exercise in exercises[k]:
+              if exercise.kind == JObject:
+                if exercise.hasKey("slug"):
+                  let slug = exercise["slug"]
+                  if slug.kind == JString:
+                    let slugStr = slug.getStr()
+                    if slugStr.len > 0:
+                      result.incl slugStr
+
+proc checkExerciseDirsAreInTrackConfig(trackDir: Path; data: JsonNode;
+                                       b: var bool; path: Path) =
+  ## Sets `b` to `false` if there is an exercise directory that is
+  ## not an exercise `slug` in `data`.
+  for exerciseKind in ["concept", "practice"]:
+    let exerciseSlugs = getExerciseSlugs(data, exerciseKind)
+    if exerciseSlugs.len > 0:
+      let exercisesDir = trackDir / "exercises" / exerciseKind
+      if dirExists(exercisesDir):
+        for exerciseDir in getSortedSubdirs(exercisesDir):
+          let dirSlug = lastPathPart(exerciseDir.string)
+          if dirSlug notin exerciseSlugs:
+            let msg = &"{q $exercisesDir} contains a directory named {q dirSlug}, " &
+                      &"which is not a `slug` in the array of {exerciseKind} exercises"
```

Comment:
I'm wondering if perhaps we should let people know that if the exercise is not yet done, they can add an entry to the `config.json` with the status set to `wip`, to prevent the exercise from showing up on the website.

Reply:
Good idea. Pushed a commit - any complains about the wording?